src/gwt/src/org/rstudio/studio/client/projects/Projects.java

```diff
@@ -605,6 +608,20 @@ public void onOpenProjectEvent(OpenProjectEvent event)
                             event.getAllowOpenInNewWindow());
    }
    
+   @Override
```

Comment:
I'm following in the footsteps of the project MRU list, e.g.

https://github.com/rstudio/rstudio/blob/eecd9388c4e97f4bf1867c80b33ef1042b038b5a/src/gwt/src/org/rstudio/studio/client/projects/ProjectMRUList.java#L102-L108

So we do indeed prompt if there are unsaved changes when attempting to switch projects.